fast64_internal/utility_anim.py

```diff
@@ -24,6 +24,19 @@ def removeTrailingFrames(frameData):
 			else:
 				break
 		frameData[i].frames = frameData[i].frames[:lastUniqueFrame + 1]
+		
+def squashFramesIfAllSame(frameData):
+	for i in range(3):
+		if len(frameData[i].frames) < 2:
+			continue
+		f0 = frameData[i].frames[0]
+		for j in range(1, len(frameData[i].frames)):
+			d = abs(frameData[i].frames[j] - f0)
+			# Allow a change of +/-1 from original frame due to rounding.
+			if d >= 2 and d != 0xFFFF:
+				break
+		else:
+			frameData[i].frames = frameData[i].frames[0:1]
```

Comment:
So huh if there's a frame identical to the first frame, only keep the first frame? What am I missing

Reply:
`else` after a `for` loop means "run this code if the loop finished by running out of iterations, rather than with break". If any frame isn't nearly identical to the first frame, it will break. Otherwise (they're all the same), it will trim to just the first frame.